### PR TITLE
Also upload javaagent to release without version

### PR DESCRIPTION
**Pull request by tylerbenson**
This will enable the following URL to point to the latest version:
https://github.com/datadog/dd-trace-java/releases/latest/download/dd-java-agent.jar

### CODE REVIEW

File: .github/workflows/add-assets-to-release.yaml [hunk 1]
```diff
@@ -20,6 +21,15 @@ jobs:
           asset_path: dd-java-agent-${{ env.VERSION }}.jar
           asset_name: dd-java-agent-${{ env.VERSION }}.jar
           asset_content_type: application/java-archive
+      - name: Upload to release unversioned
+        uses: actions/upload-release-asset@e8f9f06c4b078e705bd2ea027f0926603fc9b4d5 # 1.0.2
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          upload_url: ${{ github.event.release.upload_url }}
+          asset_path: dd-java-agent-${{ env.VERSION }}.jar
```
Comment (mar-kolya) [inline .github/workflows/add-assets-to-release.yaml:30]: I'm not familiar with this whole thing, but that `cp dd-java-agent-$VERSION.jar dd-java-agent.jar` above - shouldn't it allow to not use `env.VERSION` here? Or am I missing something?

Comment (tylerbenson) [reply .github/workflows/add-assets-to-release.yaml]: I think you're right.  I updated it.

**Review by richardstartin — COMMENTED** [review]
Can you add a flag to skip this step on patch releases? These could be identified by the version (doesn't end with `.0`)

**Review by richardstartin — APPROVED** [review]
I'm ok with this. It does require us to manually fix the latest URL after patch releases to an older than latest version, but this should be reasonably rare to make this manual step agreeable. (I'd welcome an automated solution too.)

**Review by lpriima — APPROVED** [review]

**Pull request merged**